### DISCUSSION TITLE
Legend not inheriting colormap of plot?

### DISCUSSION THREAD

**Discussion by alecloudenback** [JuliaPlots/Makie.jl#3632] [Q&A] [answered]
I'm not sure if this is an issue or errant usage, but the legend lines do not inherit the color of the lines when specified via a `colormap` and `colorrange`:

```julia
f = Figure()
ax = Axis(f[1,1])

xs = 1:100

n = 10
for i in 1:n
    lines!(ax,xs,xs .+ i*5,color = fill(i,length(xs)),colormap = :viridis, label=string(i),colorrange=(1,n))
end
f[1, 2] = Legend(f, ax, "Simulation", framevisible = false, merge=true)
f
```

produces: 

<img width="590" alt="image" src="https://github.com/MakieOrg/Makie.jl/assets/711879/85ad14e9-5ef4-4666-952e-ad7c9d86d8fa">

In this usage, how does one get the legend colors to align with the plot?

MacOS, CairoMakie v0.11.8, Makie v0.20.7

**Comment by jkrumbiegel** [comment, marked as answer]
Try passing just `i` as color, not an array. An array color will be ignored in the legend and fal back to default because it can't be shown in one symbol.
